src/core/lexer/lexer-state.js

```diff
@@ -64,7 +64,7 @@ class LexerState {
     });
 
     // Join the rules that we got above on a |, then put them all into a negative lookahead.
-    const textPattern = `((?!${rules.join('|')}).)*`;
+    const textPattern = `((?!${rules.join('|')}).)+`;
```

Comment:
I believe this was useful when I was pursuing a solution that wound up going nowhere. The tests pass either way. But this makes more sense, right? We only match text if there is some text?